include/glow/Importer/ProtobufLoader.h

```diff
@@ -105,12 +105,32 @@ class ProtobufLoader {
 
 public:
   /// \returns the node that was registered with the name \p name.
-  Node *getNodeByName(const std::string &name);
+  /// When no node has been registered for this name, the method will
+  /// either return nullptr if \p allowNullptr is true or will crash the
+  /// program otherwise. In other words, this can be used to perform
+  /// invariant checks when we know something must not be nullptr.
+  Node *getNodeByName(const std::string &name, bool allowNullptr = false) const;
 
   /// \returns the node that was registered with the name \p name or create a
   /// new Variable node for a tensor with this name.
   Node *getOrCreateNodeByName(const std::string &name);
 
+  /// Create a new node \p name initialized with \p tensor.
```

Comment:
Strictly saying, it's not a generic Node, it's always a Variable (here and in `createAndRememberNode`).
Could you please change to Variable both in names and in comments?

Reply:
Actually, I am trying to abstract that this node is a Variable :-).
Going forward, I plan to return an actual node and not a variable. I can do the change to Variable now, but I would have to undo it soonish. (Basically, I didn't want to give access to APIs not supported by Node.)

Reply:
What is plan going forward then? I don't understand how generic Node can be initialized with a tensor, but do understand how Variable can be initialized with a tensor.

Reply:
This will still init a variable with the given tensor.

The difference is that the method won't give back the variable directly but a load of the variable. (The term load is overloaded, but I don't have a better name for now.)
I.e., var vs. load(var).

Note: The plan has not been discussed yet, but if possible I would like to avoid to go back and forth on the API proper to have less code to modify if the plan is materialized. If you prefer I do the renaming of node to variable and we'll reiterate when/if the plan carry on. Or we can do the renaming if the plan is not carried on :).

Reply:
OK, let's leave tensors then and discuss your idea about loads.